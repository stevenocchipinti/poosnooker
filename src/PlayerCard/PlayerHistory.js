import React from 'react'
import styled from 'styled-components'

const colors = {
  CANNON: 'gray',
  YELLOW: 'yellow',
  GREEN: 'green',
  BLUE: 'blue',
  PINK: 'pink',
  BLACK: 'black',
  FOUL: 'red',
  POO: 'brown',
}

const Wrapper = styled.div`
  display: flex;
  justify-content: flex-end;
`
const Ball = styled.div`
  min-width: 8px;
  width: 8px;
  height: 8px;
  border-radius: 50%;
  margin: 3px;
  background: ${props => colors[props.event]};
`

export default ({history}) => {
  return (
    <Wrapper>
      {history.map((event, index) => <Ball event={event} key={index} />)}
    </Wrapper>
  )
}
