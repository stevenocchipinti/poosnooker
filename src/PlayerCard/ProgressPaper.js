import React from 'react'
import styled from 'styled-components'
import Paper from 'material-ui/Paper'
import {LinearProgress} from 'material-ui/Progress'

const StyledPaper = styled(Paper)`
  margin: 10px;
  position: relative;
`
const ProgressBar = styled(LinearProgress)`
  position: absolute;
  bottom: 0;
`

export default ({progress, children}) => (
  <StyledPaper>
    {children}
    <ProgressBar mode="determinate" value={progress * 100} />
  </StyledPaper>
)
