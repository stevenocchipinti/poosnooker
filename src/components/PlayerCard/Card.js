import React from 'react'
import styled from 'styled-components'
import Paper from 'material-ui/Paper'
import {LinearProgress} from 'material-ui/Progress'

const StyledPaper = styled(Paper)`
  margin: 10px;
  position: relative;
`

const ProgressBar = ({progress}) => (
  <div style={{position: 'absolute', bottom: 0, width: '100%'}}>
    <LinearProgress variant="determinate" value={progress * 100} />
  </div>
)

export const Card = ({progress, children, className, ...props}) => (
  <StyledPaper {...props} className={className}>
    {children}
    <ProgressBar progress={progress} />
  </StyledPaper>
)

export const SmallCard = styled(Card)`
  width: 220px;
`
