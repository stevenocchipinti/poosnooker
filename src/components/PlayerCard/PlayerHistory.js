import React from 'react'
import styled from 'styled-components'
import {colors} from '../../colors'

const Wrapper = styled.div`
  display: flex;
  justify-content: flex-end;
  height: 14px;
`
const Ball = styled.div`
  min-width: 8px;
  width: 8px;
  height: 8px;
  border-radius: 50%;
  margin: 3px;
  background: ${props => colors[props.event]};
  box-shadow: 0px 0px 1px 0px black;
`

export default ({history}) => {
  return (
    <Wrapper>
      {history.map((event, index) => <Ball event={event} key={index} />)}
    </Wrapper>
  )
}
