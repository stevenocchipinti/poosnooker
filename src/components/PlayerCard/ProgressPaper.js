import React from 'react'
import styled from 'styled-components'
import Paper from 'material-ui/Paper'
import {LinearProgress} from 'material-ui/Progress'

const variantStyle = (variant, property) => {
  const variantStyles = {
    standard: {width: '220px', height: '100px'},
    large: {width: 'auto', height: '175px'},
    horizontal: {width: 'auto', height: 'auto'},
  }
  const key = variantStyles.hasOwnProperty(variant) ? variant : 'standard'
  return variantStyles[key][property]
}

const StyledPaper = styled(Paper)`
  width: ${({variant}) => variantStyle(variant, 'width')};
  height: ${({variant}) => variantStyle(variant, 'height')};
  margin: 10px;
  position: relative;
`

export default ({progress, children, variant, ...props}) => (
  <StyledPaper variant={variant} {...props}>
    {children}
    <div style={{position: 'absolute', bottom: 0, width: '100%'}}>
      <LinearProgress variant="determinate" value={progress * 100} />
    </div>
  </StyledPaper>
)
