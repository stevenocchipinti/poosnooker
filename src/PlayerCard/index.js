import React from 'react'
import styled from 'styled-components'
import ProgressPaper from './ProgressPaper'
import PlayerHistory from './PlayerHistory'

const PlayerInfoGrid = styled.div`
  display: grid;
  padding: 10px;
  align-items: center;
  grid-row-gap: 10px;
  grid-template-areas:
    'name     score'
    'history  history';
`
const PlayerNameCell = styled.div`
  grid-area: name;
  font-weight: bold;
  font-size: 1.6em;
`
const PlayerScoreCell = styled.div`
  grid-area: score;
  justify-self: end;
`
const PlayerHistoryCell = styled.div`
  grid-area: history;
  overflow: hidden;
  display: ${props => (props.visible ? 'initial' : 'none')};
`

export default ({player}) => {
  return (
    <ProgressPaper progress={player.score / player.target}>
      <PlayerInfoGrid>
        <PlayerNameCell>{player.name}</PlayerNameCell>

        <PlayerScoreCell>
          <strong>{player.score}</strong> / {player.target}
        </PlayerScoreCell>

        <PlayerHistoryCell>
          <PlayerHistory history={player.history} />
        </PlayerHistoryCell>
      </PlayerInfoGrid>
    </ProgressPaper>
  )
}
