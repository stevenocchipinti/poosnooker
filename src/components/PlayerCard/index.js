import React from 'react'
import styled from 'styled-components'
import ProgressPaper from './ProgressPaper'
import PlayerHistory from './PlayerHistory'

const PlayerInfoGrid = styled.div`
  display: grid;
  font-size: ${({variant}) => (variant === 'large' ? '2em' : '1em')};
  padding: 20px;
  align-items: center;
  overflow: hidden;
  grid-row-gap: 10px;
  justify-content: stretch;
  grid-auto-flow: ${({variant}) =>
    variant === 'horizontal' ? 'column' : 'row'};
`

const PlayerNameCell = styled.div`
  font-weight: bold;
  font-size: 1.6em;
  justify-self: ${({variant}) =>
    variant === 'horizontal' ? 'flex-start' : 'center'};
  overflow: hidden;
`
const PlayerScoreCell = styled.div`
  justify-self: ${({variant}) =>
    variant === 'horizontal' ? 'flex-end' : 'center'};
`
const PlayerHistoryCell = styled.div`
  overflow: hidden;
`

export default ({player, variant, ...props}) => (
  <ProgressPaper
    variant={variant}
    progress={player.score / player.target}
    {...props}
  >
    <PlayerInfoGrid variant={variant}>
      <PlayerNameCell variant={variant}>{player.name}</PlayerNameCell>

      <PlayerScoreCell variant={variant}>
        <strong>{player.score}</strong> / {player.target}
      </PlayerScoreCell>

      {variant === 'large' && (
        <PlayerHistoryCell>
          <PlayerHistory history={player.history} />
        </PlayerHistoryCell>
      )}
    </PlayerInfoGrid>
  </ProgressPaper>
)
