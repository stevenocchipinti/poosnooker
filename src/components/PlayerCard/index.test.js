import React from 'react'
import renderer from 'react-test-renderer'
import 'jest-styled-components'
import PlayerCard from '.'

describe('PlayerCard', () => {
  const player = {
    name: 'Player',
    score: 10,
    target: 31,
    history: ['CANNON', 'BLUE', 'GREEN'],
  }

  describe('default variant', () => {
    it('renders correctly', () => {
      const tree = renderer.create(<PlayerCard player={player} />).toJSON()
      expect(tree).toMatchSnapshot()
    })
  })

  describe('standard variant', () => {
    it('renders correctly', () => {
      const tree = renderer
        .create(<PlayerCard variant="standard" player={player} />)
        .toJSON()
      expect(tree).toMatchSnapshot()
    })
  })

  describe('large variant', () => {
    it('renders correctly', () => {
      const tree = renderer
        .create(<PlayerCard variant="large" player={player} />)
        .toJSON()
      expect(tree).toMatchSnapshot()
    })
  })

  describe('horizontal variant', () => {
    it('renders correctly', () => {
      const tree = renderer
        .create(<PlayerCard variant="horizontal" player={player} />)
        .toJSON()
      expect(tree).toMatchSnapshot()
    })
  })
})
