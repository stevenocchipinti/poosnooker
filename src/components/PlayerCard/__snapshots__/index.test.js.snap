// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlayerCard default variant renders correctly 1`] = `
.c0 {
  width: 220px;
  height: 100px;
  margin: 10px;
  position: relative;
}

.c1 {
  display: grid;
  font-size: 1em;
  padding: 20px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  overflow: hidden;
  grid-row-gap: 10px;
  -webkit-box-pack: stretch;
  -webkit-justify-content: stretch;
  -ms-flex-pack: stretch;
  justify-content: stretch;
  grid-auto-flow: row;
}

.c2 {
  font-weight: bold;
  font-size: 1.6em;
  justify-self: center;
  overflow: hidden;
}

.c3 {
  justify-self: center;
}

<div
  className="MuiPaper-root-1 MuiPaper-elevation2-5 MuiPaper-rounded-2 c0"
  variant={undefined}
>
  <div
    className="c1"
  >
    <div
      className="c2"
    >
      Player
    </div>
    <div
      className="c3"
    >
      <strong>
        10
      </strong>
       / 
      31
    </div>
  </div>
  <div
    style={
      Object {
        "bottom": 0,
        "position": "absolute",
        "width": "100%",
      }
    }
  >
    <div
      aria-valuenow={32}
      className="MuiLinearProgress-root-28 MuiLinearProgress-colorPrimary-29"
      role="progressbar"
    >
      <div
        className="MuiLinearProgress-bar-36 MuiLinearProgress-barColorPrimary-37 MuiLinearProgress-bar1Determinate-41"
        style={
          Object {
            "transform": "scaleX(0.3225806451612903)",
          }
        }
      />
    </div>
  </div>
</div>
`;

exports[`PlayerCard horizontal variant renders correctly 1`] = `
.c0 {
  width: auto;
  height: auto;
  margin: 10px;
  position: relative;
}

.c1 {
  display: grid;
  font-size: 1em;
  padding: 20px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  overflow: hidden;
  grid-row-gap: 10px;
  -webkit-box-pack: stretch;
  -webkit-justify-content: stretch;
  -ms-flex-pack: stretch;
  justify-content: stretch;
  grid-auto-flow: column;
}

.c2 {
  font-weight: bold;
  font-size: 1.6em;
  justify-self: flex-start;
  overflow: hidden;
}

.c3 {
  justify-self: flex-end;
}

<div
  className="MuiPaper-root-1 MuiPaper-elevation2-5 MuiPaper-rounded-2 c0"
  variant="horizontal"
>
  <div
    className="c1"
  >
    <div
      className="c2"
    >
      Player
    </div>
    <div
      className="c3"
    >
      <strong>
        10
      </strong>
       / 
      31
    </div>
  </div>
  <div
    style={
      Object {
        "bottom": 0,
        "position": "absolute",
        "width": "100%",
      }
    }
  >
    <div
      aria-valuenow={32}
      className="MuiLinearProgress-root-28 MuiLinearProgress-colorPrimary-29"
      role="progressbar"
    >
      <div
        className="MuiLinearProgress-bar-36 MuiLinearProgress-barColorPrimary-37 MuiLinearProgress-bar1Determinate-41"
        style={
          Object {
            "transform": "scaleX(0.3225806451612903)",
          }
        }
      />
    </div>
  </div>
</div>
`;

exports[`PlayerCard large variant renders correctly 1`] = `
.c0 {
  width: auto;
  height: 175px;
  margin: 10px;
  position: relative;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  height: 14px;
}

.c6 {
  min-width: 8px;
  width: 8px;
  height: 8px;
  border-radius: 50%;
  margin: 3px;
  background: #ffffff;
  box-shadow: 0px 0px 1px 0px black;
}

.c7 {
  min-width: 8px;
  width: 8px;
  height: 8px;
  border-radius: 50%;
  margin: 3px;
  background: #2f4efb;
  box-shadow: 0px 0px 1px 0px black;
}

.c8 {
  min-width: 8px;
  width: 8px;
  height: 8px;
  border-radius: 50%;
  margin: 3px;
  background: #4caf50;
  box-shadow: 0px 0px 1px 0px black;
}

.c1 {
  display: grid;
  font-size: 2em;
  padding: 20px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  overflow: hidden;
  grid-row-gap: 10px;
  -webkit-box-pack: stretch;
  -webkit-justify-content: stretch;
  -ms-flex-pack: stretch;
  justify-content: stretch;
  grid-auto-flow: row;
}

.c2 {
  font-weight: bold;
  font-size: 1.6em;
  justify-self: center;
  overflow: hidden;
}

.c3 {
  justify-self: center;
}

.c4 {
  overflow: hidden;
}

<div
  className="MuiPaper-root-1 MuiPaper-elevation2-5 MuiPaper-rounded-2 c0"
  variant="large"
>
  <div
    className="c1"
  >
    <div
      className="c2"
    >
      Player
    </div>
    <div
      className="c3"
    >
      <strong>
        10
      </strong>
       / 
      31
    </div>
    <div
      className="c4"
    >
      <div
        className="c5"
      >
        <div
          className="c6"
        />
        <div
          className="c7"
        />
        <div
          className="c8"
        />
      </div>
    </div>
  </div>
  <div
    style={
      Object {
        "bottom": 0,
        "position": "absolute",
        "width": "100%",
      }
    }
  >
    <div
      aria-valuenow={32}
      className="MuiLinearProgress-root-28 MuiLinearProgress-colorPrimary-29"
      role="progressbar"
    >
      <div
        className="MuiLinearProgress-bar-36 MuiLinearProgress-barColorPrimary-37 MuiLinearProgress-bar1Determinate-41"
        style={
          Object {
            "transform": "scaleX(0.3225806451612903)",
          }
        }
      />
    </div>
  </div>
</div>
`;

exports[`PlayerCard standard variant renders correctly 1`] = `
.c0 {
  width: 220px;
  height: 100px;
  margin: 10px;
  position: relative;
}

.c1 {
  display: grid;
  font-size: 1em;
  padding: 20px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  overflow: hidden;
  grid-row-gap: 10px;
  -webkit-box-pack: stretch;
  -webkit-justify-content: stretch;
  -ms-flex-pack: stretch;
  justify-content: stretch;
  grid-auto-flow: row;
}

.c2 {
  font-weight: bold;
  font-size: 1.6em;
  justify-self: center;
  overflow: hidden;
}

.c3 {
  justify-self: center;
}

<div
  className="MuiPaper-root-1 MuiPaper-elevation2-5 MuiPaper-rounded-2 c0"
  variant="standard"
>
  <div
    className="c1"
  >
    <div
      className="c2"
    >
      Player
    </div>
    <div
      className="c3"
    >
      <strong>
        10
      </strong>
       / 
      31
    </div>
  </div>
  <div
    style={
      Object {
        "bottom": 0,
        "position": "absolute",
        "width": "100%",
      }
    }
  >
    <div
      aria-valuenow={32}
      className="MuiLinearProgress-root-28 MuiLinearProgress-colorPrimary-29"
      role="progressbar"
    >
      <div
        className="MuiLinearProgress-bar-36 MuiLinearProgress-barColorPrimary-37 MuiLinearProgress-bar1Determinate-41"
        style={
          Object {
            "transform": "scaleX(0.3225806451612903)",
          }
        }
      />
    </div>
  </div>
</div>
`;
